### ISSUE TITLE
[BUG] focus is removed from modal when navigating to new page or component in modal flows

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#3896]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
Main problem is described https://github.com/kirbydesign/designsystem/issues/3801, but with some additional context added here:

This is the case for any modal where the active element disappears, not only routed modals - e.g. a "switch" setup that changes the component inside the active modal on next/prev buttons in a modal footer.

Those instances might be harder to support out of the box, as there is no navigation event, and anything might happen on a button click. But maybe we can provide a helper to refocus the modal, that can be called in a click-handler?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/develop/.github/SUPPORT.md#team-kirby); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by RasmusKjeldgaard** [comment]
This is the case for any modal where the active element disappears - e.g. a "switch" setup that changes the component inside the active modal on next/prev buttons in a modal footer.

Those instances might be harder to support out of the box, as there is no navigation event, and anything might happen on a button click. But maybe we can provide a helper to refocus the modal, that can be called in a click-handler?

**Comment by Fuzzy3** [comment]
The idea of having a helper for those cases where they use a switch-case modal and not a routed modal, is something we definitely should look into, since its been asked for.
